test/openssl/test_ssl.rb

```diff
@@ -2062,6 +2062,52 @@ def test_get_sigalg
     end
   end
 
+  def test_pqc_sigalg
+    # Post-quantum cryptography is supported on OpenSSL 3.5 or later.
+    return unless openssl?(3, 5, 0)
+
+    mldsa = Fixtures.pkey("mldsa65")
+    mldsa_ca_key  = Fixtures.pkey("mldsa-1")
+    mldsa_ca_cert = issue_cert(@ca, mldsa_ca_key, 1, @ca_exts, nil, nil,
+                               digest: nil)
+    mldsa_cert = issue_cert(@svr, mldsa, 60, [], mldsa_ca_cert, mldsa_ca_key,
+                            digest: nil)
+    rsa = Fixtures.pkey("rsa2048")
+    rsa_cert = issue_cert(@svr, rsa, 61, [], @ca_cert, @ca_key)
+    ctx_proc = -> ctx {
+      # Unset values set by start_server
+      ctx.cert = ctx.key = ctx.extra_chain_cert = nil
+      ctx.add_certificate(mldsa_cert, mldsa)
+      ctx.add_certificate(rsa_cert, rsa)
+    }
+
+    server_proc = -> (ctx, ssl) {
+      assert_equal('mldsa65', ssl.sigalg)
+
+      readwrite_loop(ctx, ssl)
+    }
+    start_server(ctx_proc: ctx_proc, server_proc: server_proc) do |port|
+      server_connect(port) { |ssl|
+        assert_equal('mldsa65', ssl.peer_sigalg)
+        ssl.puts "abc"; ssl.gets
+      }
```

Comment:
The client should set `SSLContext#sigalgs=` to specify the preference order. OpenSSL 3.5 may prefer ML-DSA by default, but it is affected by the system OpenSSL configuration.

Reply:
OK. That makes sense. Thank you for catching the issue! I added the line `ctx.sigalgs = 'mldsa65'` with the comment about why it is needed.